app/src/test/java/android/util/Log.java

```diff
@@ -0,0 +1,28 @@
+package android.util;
+
+// This is needed for the unit tests to run when the functions called include debug messages. (Mocking)
+public class Log {
```

Comment:
I would suggest we add the following in our build.gradle ->
```
 testOptions { 
    unitTests.returnDefaultValues = true
  }
```
Otherwise we will have to add also a lot of other classes like this in the future whereas the gradle option is very convenient. See also: https://sites.google.com/a/android.com/tools/tech-docs/unit-testing-support#TOC-Method-...-not-mocked.-

Reply:
Thanks, I have changed it.